service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -89,26 +81,19 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             caseData.getAllRespondents(), caseDataBefore.getAllRespondents());
 
         caseDetails.getData().put(RESPONDENTS_KEY, respondentService.removeHiddenFields(respondents));
+        if (!OPEN.equals(caseData.getState()) && featureToggleService.hasRSOCaseAccess()) {
+            caseDetails.getData().putAll(respondentPolicyService.generateForSubmission(caseDetails));
+        }
```

Comment:
this does not looks right, respondentPolicyService.generateForSubmission(caseDetails) will override what was there and NoC history will be lost

Reply:
Discussed with @tomaszpowroznik. As part of https://tools.hmcts.net/jira/browse/FPLA-2830 there will be a separate audit of the change in organisation. I've asked Nigel, and waiting for a response in case previousOrganisation has some functional meaning, or it's just a substitute for audit. That is anyway OK for respondent journey, and we can in case modify the generation method before going live with NOC

Reply:
Nigel  12:47 PM
The previous organisation was added so that it could be displayed via ExUI or other user interfaces.  There’s no requirement on services to do anything with it at all - it gets created/maintained via the NoC functionality.